hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -422,11 +425,21 @@ private Table newHmsTable(TableMetadata metadata) {
     Preconditions.checkNotNull(metadata, "'metadata' parameter can't be null");
     final long currentTimeMillis = System.currentTimeMillis();
 
+    String defaultUser;
+    try {
```

Comment:
Actually there is one thing: I think in case the HMS_TABLE_OWNER is set we don't have to query the UGI.getCurrentUser() because it won't be used anyway.

Reply:
Ack, taken